packages/core/src/lib/GooglePlay.ts

```diff
@@ -62,12 +63,104 @@ export class GooglePlay {
    * https://github.com/Triple-T/gradle-play-publisher#uploading-a-pre-existing-artifact
    * @param track - Specifies the track that the user would like to publish to.
    */
-  async publishBundle(track: PlayStoreTrack, filepath: string): Promise<void> {
+  async publishBundleWithGradle(track: PlayStoreTrack, filepath: string): Promise<void> {
     // Uploads the artifact to the default internal track.
     await this.gradleWrapper.executeGradleCommand(
         ['publishBundle', '--artifact-dir', filepath, '--track', track]);
   }
 
+  /**
+   * This calls the publish bundle command and publishes an existing artifact to Google
+   * Play.
+   * Calls the following Play API commands in order:
+   * Edits.Insert
+   * Edits.Bunldes.Upload
+   * Edits.Tracks.Update
+   * Edits.Commit
+   * https://developers.google.com/android-publisher/api-ref/rest/v3/edits.bundles/upload
+   * https://developers.google.com/android-publisher/edits#workflow
+   * @param track - Specifies the track that the user would like to publish to.
+   * @param filepath - Filepath of the App bundle you would like to upload.
+   * @param packageName - packageName of the bundle.
+   * @param retainedBundles - all bundles that should be retained on upload. This is useful for
+   *   ChromeOS only releases.
+   */
+  async publishBundle(
+      track: PlayStoreTrack,
+      filepath: string,
+      packageName: string,
+      retainedBundles: number[],
+  ): Promise<void> {
+    // TODO(@nohe427): Remove this check when refactor is finished.
+    if (!this._googlePlayApi) {
```

Comment:
nit: I know this is going to be removed, but maybe throw an error or a log message in the meantime?

Reply:
I think throwing an error is fine here - it doesn't need to be tidy since users won't come across it, but it makes it obvious what went wrong if a developer hits it.